packages/mars/src/options/config.ts

```diff
@@ -0,0 +1,83 @@
+import { providers, Wallet } from 'ethers'
+import Ganache from 'ganache-core'
+import { ExecuteOptions } from '../execute/execute'
+import { createJsonInputs } from '../verification'
+import { exit } from './checks'
+import { getCommandLineOptions } from './cli'
+import { getDefaultOptions } from './defaults'
+import { getEnvironmentOptions } from './environment'
+import { Options } from './Options'
+
+export async function getConfig(options: Options): Promise<ExecuteOptions> {
+  const merged = {
+    ...getDefaultOptions(),
+    ...getEnvironmentOptions(),
+    ...getCommandLineOptions(),
+    ...options,
```

Comment:
Shouldn't it be CLI > env > code > defaults?

Reply:
Yeah, I am actually not sure about this. @Marik-D @barrutko ?

Reply:
I still somehow feel, that code should be superior to everything else and implemented order makes sense. If you hardcode some values in the script, you shouldn't be able to override them with CLI args. IMO.

Reply:
+1 @vanruch

Reply:
Regardless of what order we pick here, we should print the config for the user after merging. As it looks like merge semantics aren't trivial and can be confusing

Reply:
@Marik-D printing the config results in leaking the private key